server/lib/page-type-util.js

```diff
@@ -0,0 +1,126 @@
+
+const templateFileToPageTypeMap = {
+    'pages/page': PageTypes.PAGE,
+    'pages/product': PageTypes.PRODUCT,
+    'pages/category': PageTypes.CATEGORY,
+    'pages/brand': PageTypes.BRAND,
+    'pages/account/return-saved': PageTypes.ACCOUNT_RETURN_SAVED,
+    'pages/account/add-return': PageTypes.ACCOUNT_ADD_RETURN,
+    'pages/account/returns': PageTypes.ACCOUNT_RETURNS,
+    'pages/account/add-address': PageTypes.ACCOUNT_ADD_ADDRESS,
+    'pages/account/add-wishlist': PageTypes.ACCOUNT_ADD_WISHLIST,
+    'pages/account/wishlists': PageTypes.ACCOUNT_WISHLISTS,
+    'pages/account/wishlist-details': PageTypes.ACCOUNT_WISHLIST_DETAILS,
+    'pages/account/edit': PageTypes.ACCOUNT_EDIT,
+    'pages/account/addresses': PageTypes.ACCOUNT_ADDRESS,
+    'pages/account/inbox': PageTypes.ACCOUNT_INBOX,
+    'pages/account/download-item': PageTypes.ACCOUNT_DOWNLOAD_ITEM,
+    'pages/account/orders/all': PageTypes.ACCOUNT_ORDERS_ALL,
+    'pages/account/orders/invoice': PageTypes.ACCOUNT_ORDERS_INVOICE,
+    'pages/account/orders/details': PageTypes.ACCOUNT_ORDERS_DETAILS,
+    'pages/account/orders/completed': PageTypes.ACCOUNT_ORDERS_COMPLETED,
+    'pages/account/recent-items': PageTypes.ACCOUNT_RECENT_ITEMS,
+    'pages/auth/account-created': PageTypes.AUTH_ACCOUNT_CREATED,
+    'pages/auth/login': PageTypes.AUTH_LOGIN,
+    'pages/auth/create-account': PageTypes.AUTH_CREATE_ACC,
+    'pages/auth/forgot-password': PageTypes.AUTH_FORGOT_PASS,
+    'pages/auth/new-password': PageTypes.AUTH_NEW_PASS,
+    'pages/blog-post': PageTypes.BLOG_POST,
+    'pages/blog': PageTypes.BLOG,
+    'pages/brands': PageTypes.BRANDS,
+    'pages/cart': PageTypes.CART,
+    'pages/compare': PageTypes.COMPARE,
+    'pages/contact-us': PageTypes.CONTACT_US,
+    'pages/home': PageTypes.HOME,
+    'pages/gift-certificate/purchase': PageTypes.GIFT_CERT_PURCHASE,
+    'pages/gift-certificate/redeem': PageTypes.GIFT_CERT_REDEEM,
+    'pages/gift-certificate/balance': PageTypes.GIFT_CERT_BALANCE,
+    'pages/order-confirmation': PageTypes.ORDER_INFO,
+    'pages/search': PageTypes.SEARCH,
+    'pages/sitemap': PageTypes.SITEMAP,
+    'pages/subscribed': PageTypes.SUBSCRIBED,
+    'pages/unsubscribe': PageTypes.UNSUBSCRIBE,
+};
+
+const pageTypeWithEntityId = [
+    PageTypes.BLOG_POST,
+    PageTypes.BRAND,
+    PageTypes.CATEGORY,
+    PageTypes.CONTACT_US,
+    PageTypes.PAGE,
+    PageTypes.PRODUCT,
+];
+
+/**
+ * Convert a templateFile to pageType
+ *
+ * @param {string} templateFile
+ * @returns {string | null}
+ */
+function getPageType(templateFile) {
+    const pageType = templateFileToPageTypeMap[templateFile];
+
+    return pageType;
+}
+
+/**
+ * Check if templateFile must have entityId
+ *
+ * @param {string} pageType
+ * @param {number} entityId
+ * @returns {boolean}
+ */
+function isValidPageTypeEntityIdReq(pageType, entityId) {
+    if (typeof entityId !== 'number') {
+        return false;
+    }
+
+    return pageTypeWithEntityId.indexOf(pageType) >= 0;
```

Comment:
Is this actually necessary? Can we not just check if the entityId is present and just trust that it's correct rather than checking another list we have to maintain here?